plugins/modules/cloudflare_dns.py

```diff
@@ -757,12 +780,36 @@ def ensure_dns_record(self, **kwargs):
             }
             search_value = str(params['cert_usage']) + '\t' + str(params['selector']) + '\t' + str(params['hash_type']) + '\t' + params['value']
 
+        if params['type'] == 'CAA':
+            for attr in [params['flag'], params['tag'], params['value']]:
+                if (attr is None) or (attr == ''):
+                    self.module.fail_json(msg="You must provide flag, tag and a value to create this record type")
```

Comment:
Hasn't this already been checked by the other code further below?

Reply:
Yes, I think these checks are useless, but they are there for the other types, so I took the code style from the existing module.
I think we could keep these checks to avoid code inconsistency.
Then, maybe we could open another PR to remove useless code.

Reply:
Sounds OK for me.